messaging/src/main/java/org/axonframework/eventhandling/deadletter/DeadLetteredEventProcessingTask.java

```diff
@@ -89,15 +94,30 @@ public EnqueueDecision<EventMessage<?>> process(DeadLetter<? extends EventMessag
                   .put(DeadLetter.class.getName(), letter);
         unitOfWork.onPrepareCommit(uow -> decision.set(onCommit(letter)));
         unitOfWork.onRollback(uow -> decision.set(onRollback(letter, uow.getExecutionResult().getExceptionResult())));
-        unitOfWork.executeWithResult(() -> handle(letter));
+        unitOfWork.executeWithResult(() -> handleWithInterceptors(letter, unitOfWork));
 
         return ObjectUtils.getOrDefault(decision.get(), Decisions::ignore);
     }
 
-    private Object handle(DeadLetter<? extends EventMessage<?>> letter) throws Exception {
+    private void handle(EventMessage<?> eventMessage) throws Exception {
         for (EventMessageHandler handler : eventHandlingComponents) {
-            handler.handle(letter.message());
+            handler.handle(eventMessage);
         }
+    }
+
+    private Object handleWithInterceptors(
+            DeadLetter<? extends EventMessage<?>> letter,
+            UnitOfWork<? extends EventMessage<?>> unitOfWork
+    ) throws Exception {
+        unitOfWork.transformMessage(m -> letter.message());
```

Comment:
I may be wrong, but I think the `EventMessage` is already present in the `UnitOfWork`.
Line 90 set's the `UnitOfWork` with the letter in question.
Hence, I think we drop the `transformMessage` invocation.

Reply:
Granted, I can imagine you added the line on purpose. So, if I'm wrong here, I would like to understand why ^_^

Reply:
If we use the same interceptors, they might expect the message to be an event message, not a dead letter message. It does remove some of the dead letter-specific things. So we probably want to either directly put just the event message there or keep the dead letter message, and leave it to end users to make sure the interceptors can also handle dead letter messages if needed. I'm not sure what's wise, maybe we can have a quick call?

Reply:
Never mind, missed that the actual event message was already the one set. So removing the line.

Reply:
Removed in [20ca823](https://github.com/AxonFramework/AxonFramework/pull/2661/commits/20ca823354eeaff5b50b4f53b77ba63e45904588)